docs/proposals/PROPOSAL_C.md

```diff
@@ -0,0 +1,323 @@
+
+This is an example of using a Node manifest to describe a single ingredient:
+
+```jsonc
+{
+  "schemaVersion": 3,
+  "mediaType": "application/vnd.oci.node.v1+json",
+  "description": "Ice Cream Cone",
+  "objects": [
+    {
+      "mediaType": "cones/vnd.waffleconeinc.v0.1.0+cone",
+      "size": 10,
+      "digest": "sha256:114fff13c0"
+    }
+  ]
+  "reference": {}
+}
+```
+
+This is an example of creating a tree structure with the node manifest:
+
+Existing ice creams:
+
+```jsonc
+{
+  "schemaVersion": 3,
+  "mediaType": "application/vnd.oci.node.v1+json",
+  "description": "Vanilla Ice Cream - created two months ago",
+  "objects": [
+    {
+      "mediaType": "icecream/vnd.glenandlarry.vanilla.v2.0.1+batch",
+      "size": 12,
+      "digest": "sha256:4a77111a"
+    }
+  ]
+  "reference": {}
+}
+
+{
+  "schemaVersion": 3,
+  "mediaType": "application/vnd.oci.node.v1+json",
+  "description": "Chocolate Ice Cream - created one month ago",
+  "objects": [
+    {
+      "mediaType": "icecream/vnd.glenandlarry.chocolate.v0.0.2+batch",
+      "size": 10,
+      "digest": "sha256:c0c01a7e"
+    }
+  ]
+  "reference": {}
+}
+```
+
+New ice cream:
+
+```jsonc
+{
+  "schemaVersion": 3,
+  "mediaType": "application/vnd.oci.node.v1+json",
+  "description": "Vanilla Chocolate Swirl - created two days ago",
+  "objects": [
+    {
+      "mediaType": "application/vnd.oci.node.v1+json",
+      "size": 15,
+      "digest": "sha256:b1ab1ab1a"
+    },
+    {
+      "mediaType": "application/vnd.oci.node.v1+json",
+      "size": 12,
+      "digest": "sha256:cabba9ebeef"
+    }
+  ]
+  "reference": {}
+}
+
+```
+
+### Registry HTTP API
+
+No modifications to the existing API are needed. The proposal includes the following new APIs
+
+#### References to a tag or digest
+
+```
+GET /v3/<name>/referrers/<reference>
+```
+
+Response:
+
+```jsonc
+{
+  "referrers":
+    [list of descriptors]
+}
+```
+
+Example:
+
+```
+GET /v3/glenandlarry/referrers/sha256:bec12a1c12a1
+```
+
+Gives response:
+
+```jsonc
+{
+  "referrers": [
+    {
+      "mediaType": "icecream/vnd.glenandlarry.chocolate.v0.0.2+batch",
+      "size": 10,
+      "digest": "sha256:c0c01a7e"
+    },
+    {
+      "mediaType": "icecream/vnd.glenandlarry.vanilla.v2.0.1+batch",
+      "size": 12,
+      "digest": "sha256:4a77111a"
+    },
+    {
+      "mediaType": "icecream/vnd.onlyberry.strawberry.v0.0.1+batch",
+      "size": 12,
+      "digest": "sha256:5d12abe12121"
+    },
+  ]
+
+```
+
+#### Reference of a tag or digest
+
+```
+GET /v3/<name>/reference/<reference>
+```
+
+Response:
+
+```jsonc
+{
+  descriptor OR empty
+}
+```
+
+Example:
+
+```
+GET /v3/glenandlarry/reference/sha256:abcdef123450
+```
+
+Gives response:
+
+```jsonc
+{
+  "mediaType": "icecream/vnd.glenandlarry.chocolatesprinkles.v2.0.0+package",
+  "size": 1024,
+  "digest": "sha256:1cec12ea77"
+  
+}
+```
+
+#### Objects contained in a node manifest
```

Comment:
Do we need to use the objects API, or could the existing blob and manifest APIs be used? If the latter, how would clients know which of the two sets of APIs to use? I'm thinking of this from a client that wants to copy images between registries, or some other higher level activity,  but that doesn't know the new media types.